src/faebryk/libs/exceptions.py

```diff
@@ -13,33 +11,42 @@
 class UserException(Exception):
     """A user-caused exception."""
 
-    # TODO: Add interface for getting user-facing exception information
+    # TODO: Add / refine interface for getting user-facing exception information
     # - Origin?
     # - Title?
     # - Description?
     # - Suggestions?
     # - Help text?
     # __print__?
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        *args,
+        title: str | None = None,
+        **kwargs,
+    ) -> None:
         super().__init__(*args, **kwargs)
+        self.message = args[0] if args else ""
+        self._title = title
 
+    @property
+    def title(self):
+        """Return the name of this error, without the "User" prefix."""
+        if self._title is not None:
+            return self._title
 
-class UserResourceException(UserException):
-    """Indicates an issue with a user-facing resource, e.g. layout files."""
-
+        error_name = self.__class__.__name__
+        return error_name.removeprefix("User")
 
-def in_debug_session() -> ModuleType | None:
-    """
-    Return the debugpy module if we're in a debugging session.
-    """
-    if "debugpy" in sys.modules:
-        import debugpy
+    def get_frozen(self) -> tuple:
+        """
+        Return a frozen version of this error.
+        """
+        return (self.__class__, self.message, self._title)
 
-        if debugpy.is_client_connected():
-            return debugpy
 
-    return None
+class UserResourceException(UserException):
+    """Indicates an issue with a user-facing resource, e.g. layout files."""
 
```

Comment:
This is a very handy and, imo low-level util. It probably belongs here more than `errors`